File: admin/src/api/quotaPackage.ts
import { client } from './client'
import type { ApiResponse, PagedResponse } from '../types/api'

export interface QuotaPackage {
  id: number
  name: string
  quota_value: number
  price_cents: number
  currency: string
  status: number
  sort_order: number
  created_at: string
  updated_at: string
}

export interface QuotaPackageForm {
  name: string
  quota_value: number
  price_cents: number
  currency: string
  sort_order: number
}

export function listQuotaPackages(page = 1, pageSize = 20, keyword?: string) {
  let url = `/quota-packages?page=${page}&page_size=${pageSize}`
  if (keyword) url += `&keyword=${encodeURIComponent(keyword)}`
  return client.get<ApiResponse<PagedResponse<QuotaPackage>>, ApiResponse<PagedResponse<QuotaPackage>>>(url)
}

export function createQuotaPackage(data: QuotaPackageForm) {
  return client.post<ApiResponse<QuotaPackage>, ApiResponse<QuotaPackage>>('/quota-packages', data)
}

export function updateQuotaPackage(id: number, data: Partial<QuotaPackageForm> & { status?: number }) {
  return client.put<ApiResponse<QuotaPackage>, ApiResponse<QuotaPackage>>(`/quota-packages/${id}`, data)
}

export function deleteQuotaPackage(id: number) {
  return client.delete<ApiResponse<unknown>, ApiResponse<unknown>>(`/quota-packages/${id}`)
}

export function batchUpdateQuotaPackageStatus(ids: number[], status: number) {
  return client.post<ApiResponse<null>, ApiResponse<null>>('/quota-packages/batch-status', { ids, status })
}

export function batchDeleteQuotaPackage(ids: number[]) {
  return client.post<ApiResponse<null>, ApiResponse<null>>('/quota-packages/batch-delete', { ids })
}

File: server/internal/domain/quota_package.go
package domain

import "time"

type QuotaPackageStatus int

const (
	QuotaPackageDisabled QuotaPackageStatus = 0
	QuotaPackageEnabled  QuotaPackageStatus = 1
)

type QuotaPackage struct {
	ID         uint64             `gorm:"primaryKey" json:"id"`
	Name       string             `gorm:"size:128;not null" json:"name"`
	QuotaValue int64              `gorm:"not null" json:"quota_value"`
	PriceCents int64              `gorm:"not null" json:"price_cents"`
	Currency   string             `gorm:"size:8;not null;default:'CNY'" json:"currency"`
	Status     QuotaPackageStatus `gorm:"not null;default:1" json:"status"`
	SortOrder  int                `gorm:"not null;default:0" json:"sort_order"`
	CreatedAt  time.Time          `json:"created_at"`
	UpdatedAt  time.Time          `json:"updated_at"`
}

func (QuotaPackage) TableName() string { return "quota_packages" }
